third_party/freedreno/wrap/wrap-util.c

```diff
@@ -35,12 +69,67 @@ static uint64_t chip_id = 0;
 static pthread_mutex_t l = PTHREAD_RECURSIVE_MUTEX_INITIALIZER;
 #endif
 
+static pthread_mutex_t write_lock = PTHREAD_MUTEX_INITIALIZER;
+
 char *getcwd(char *buf, size_t size);
 
 int __android_log_print(int prio, const char *tag,  const char *fmt, ...);
 
 static char tracebuf[4096], *tracebufp = tracebuf;
 
+static struct device_file *get_file(int device_fd)
+{
+	if (device_fd == -1)
+		return &device_files[0];
+
+	for (int i = 0; i < MAX_DEVICE_FILES; i++) {
+		struct device_file *df = &device_files[i];
+		if (device_fd == df->device_fd)
+			return df;
+	}
+	return NULL;
+}
+
+static struct device_file *add_file(int device_fd)
+{
+	if (device_fd == -1)
+		return &device_files[0];
+
+	for (int i = 0; i < MAX_DEVICE_FILES; i++) {
+		struct device_file *df = &device_files[i];
+		if (df->device_fd == -1) {
+			df->device_fd = device_fd;
+			// GOOGLE: Add debug log.
```

Comment:
Maybe we should find a better way to track the changes. my worry is that for people who are doing integrations in the future, it is very likely to miss those google specific changes. any thoughts on this? (it is ok with this MR, just thinking about future :) )

Reply:
would it be cleaner to apply Google specific changes as a separate commit?

Reply:
Thanks for separating the commits. Are you able to keep this commit separated while merging back to main?

Reply:
thanks, let me know if you need me change any setting of the project.